### PR TITLE
fix: make global + column filtering work in Tables (@okta/odyssey-react-mui/labs)

### PR DESCRIPTION
**Pull request by benschell-okta**
### Description

Pass in the stored `globalFilter` and `columnFilters` into state to make filtering work as expected.

### CODE REVIEW

**Review by KevinGhadyani-Okta — CHANGES_REQUESTED** [review]
How do we want to set state and initialState based on the user's inputs? I would assume we want the user to be able to override our settings.

**Comment by benschell-okta** [comment]
> How do we want to set state and initialState based on the user's inputs? I would assume we want the user to be able to override our settings.

This implies flipping the ordering in PaginatedTable as well, right? Where pagination props are also passed in.

**Review by KevinGhadyani-Okta — APPROVED** [review]

File: packages/odyssey-react-mui/src/labs/PaginatedTable.tsx [hunk 1]
```diff
@@ -194,14 +194,23 @@ const PaginatedTable = <TData extends DefaultMaterialReactTableData>({
...
     () => ({
-      ...state,
       pagination: {
         pageIndex: pagination.pageIndex,
         pageSize: pagination.pageSize,
       },
       rowSelection,
+      globalFilter,
+      columnFilters,
+      ...state,
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/PaginatedTable.tsx:204]: Do we also want this for `initialState`?

Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/PaginatedTable.tsx:204]: Discussed separately, but for posterity: it is already in the preferred order: https://github.com/okta/odyssey/blob/4c482302d8ca74fa8566afcf66989064dd397d8a/packages/odyssey-react-mui/src/labs/PaginatedTable.tsx#L187-L193

**Pull request merged**